### PR TITLE
change primary hero gradients to be photon color names instead of hex colors

### LINKED ISSUE #12132: Send the photon color names for gradient for the Primary Hero shelf
In order to avoid having to use inline styles in the front-end for `linear-gradient`, and also to support the use of the `:hover` pseudo-selector for the cta button in the Primary Hero shelf, it would be better if the hero API could provide information that can be used to determine a `className` defined in the front-end.

We discussed that if the hero API returns the names of photon colors as the values for `start` and `end` for `gradient`, then the front-end can concatenate those into a `className` which would include those colors in its definition. This would allow for some synchronization of the colors between  server and front-end without having to manually keep them in sync.

The bottom line of the change is this:

Instead of sending hex colors for  `start` and `end` for `gradient`, send the corresponding photon color names.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #12132

### CODE REVIEW

File: docs/topics/api/hero.rst [hunk 1]
```diff
@@ -41,8 +41,8 @@ small number of shelves this endpoint is not paginated.
     :query string lang: Activate translations in the specific language for that query. (See :ref:`translated fields <api-overview-translations>`)
     :>json array results: The array containing the results for this query.
     :>json object results[].gradient: The background colors used for the gradient.
-    :>json string results[].gradient.start: The starting color for gradient - typically top or left.
-    :>json string results[].gradient.end: The ending color for gradient - typically bottom or right.
+    :>json string results[].gradient.start: The starting color name for gradient - typically top or left. The names are the photon color variables from https://github.com/FirefoxUX/photon-colors/blob/master/photon-colors.scss
```
Comment (bobsilverberg) [inline docs/topics/api/hero.rst]: Nit: Should be "the name is from", not "the names are", as this is listed once for each name.

File: docs/topics/api/hero.rst [hunk 2]
```diff
@@ -41,8 +41,8 @@ small number of shelves this endpoint is not paginated.
...
+    :>json string results[].gradient.end: The ending color name for gradient - typically bottom or right. The names are the photon color variables from https://github.com/FirefoxUX/photon-colors/blob/master/photon-colors.scss
```
Comment (EnTeQuAk) [inline docs/topics/api/hero.rst]: Given its restructuredtext you can use a link and a description instead too

File: docs/topics/api/hero.rst [hunk 3]
```diff
@@ -41,8 +41,8 @@ small number of shelves this endpoint is not paginated.
     :query string lang: Activate translations in the specific language for that query. (See :ref:`translated fields <api-overview-translations>`)
     :>json array results: The array containing the results for this query.
     :>json object results[].gradient: The background colors used for the gradient.
-    :>json string results[].gradient.start: The starting color for gradient - typically top or left.
-    :>json string results[].gradient.end: The ending color for gradient - typically bottom or right.
+    :>json string results[].gradient.start: The starting color name for gradient - typically top or left. The name is from the `photon color variables <https://github.com/FirefoxUX/photon-colors/blob/master/photon-colors.scss>`_.
+    :>json string results[].gradient.end: The ending color name for gradient - typically bottom or right. The name is from the `photon color variables <https://github.com/FirefoxUX/photon-colors/blob/master/photon-colors.scss>`_.
```
Comment (eviljeff) [inline docs/topics/api/hero.rst]: There's something weird happening with the RST -> html rendering meaning there's an extra `<p>` inserted in the middle of the gradient.end line after the `:`.  If either one of the links is broken/removed then there isn't an extra `<p>` - it only occurs when both of them are in the same list of parameters/response.

Screenshot 2019-08-19 18 00 17

**Comment by eviljeff** [comment]
I'm going to merge this despite the docs formatting weirdness but it'd be good to work out how to fix it anyway.

**Pull request merged**